### PR TITLE
feat: support load code from local file for issue 59

### PR DESCRIPTION
**Pull request by poppinlp**
This RP is for #59. The main logic change is that `showProblem` operation will try to load local code first. Here are the changes:

- Add `lodash.kebabcase` in dependencies to generate a slug
- Add `langExt` in `shared.ts` to do the extension mapping for language
- Add `problemUtils.ts` which contains utils for problem
- The `showProblem` in `leetCodeExecutor.ts` will return the code file path right now

### CODE REVIEW

File: src/utils/problemUtils.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import kebabCase = require("lodash.kebabcase");
+import { IProblem, langExt } from "../shared";
+
+export function genFileName(node: IProblem, language: string): string {
+    const name: string = kebabCase(node.name);
+    const ext: string | undefined = langExt.get(language);
+    return `${node.id}.${name}.${ext}`;
```
Comment (jdneo) [inline src/utils/problemUtils.ts]: `ext` here might be `undefined`. Though according to current logic, it's impossible.

I think you can have another util method here to parse the `ext` from a language name. Meanwhile, throw an error if the language is not supported.

File: src/utils/problemUtils.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import kebabCase = require("lodash.kebabcase");
```
Comment (jdneo) [inline src/utils/problemUtils.ts]: Use `import` instead of `require`.

File: src/leetCodeExecutor.ts [hunk 3]
```diff
@@ -74,8 +75,17 @@ class LeetCodeExecutor {
         );
     }
 
-    public async showProblem(id: string, language: string, outDir: string): Promise<string> {
-        return await this.executeCommandWithProgressEx("Fetching problem data...", "node", [await this.getLeetCodeBinaryPath(), "show", id, "-gx", "-l", language, "-o", `"${outDir}"`]);
+    public async showProblem(node: IProblem, language: string, outDir: string): Promise<string> {
+        const fileName: string = genFileName(node, language);
+        const filePath: string = path.join(outDir, fileName);
+        const hasLocalCode: boolean = await fse.pathExists(filePath);
```
Comment (jdneo) [inline src/leetCodeExecutor.ts]: `hasLocalCode` is redundant. We can move `await fse.pathExists(filePath)` into the if block.

File: src/utils/problemUtils.ts (hunk 2 above)
Comment (poppinlp) [reply src/utils/problemUtils.ts]: I use it in this way because of the issue of lodash. T_T

File: src/utils/problemUtils.ts (hunk 1 above)
Comment (poppinlp) [reply src/utils/problemUtils.ts]: OK. I'll fix it.

File: src/leetCodeExecutor.ts (hunk 3 above)
Comment (poppinlp) [reply src/leetCodeExecutor.ts]: OK.

File: src/utils/problemUtils.ts (hunk 2 above)
Comment (jdneo) [reply src/utils/problemUtils.ts]: OK, it's fine.

**Comment by poppinlp** [comment]
I'm sorry for the lint error. T_T

**Comment by jdneo** [comment]
@poppinlp Thank you for the contribution. Nice job!

I'll send out another PR to add you the contributors..

**Pull request merged**